src/store/modules/resources.js

```diff
@@ -20,21 +26,68 @@ export default {
       { key: "dimensionX", title: "dimensionX" },
       { key: "dimensionY", title: "dimensionY" },
       { key: "dimensionZ", title: "dimensionZ" },
+      { key: "description", title: "Description" },
+      { key: "delete", title: "", slot: "delete" },
+      { key: "edit", title: "", slot: "edit" },
     ],
   }),
   mutations: {
     setResources(state, resources) {
       state.resources = resources;
     },
+    addResources(state, payload) {
+      state.resources.push(payload);
+    },
+    removeResource(state, payload) {
+      state.resources = state.resources.filter(
+        (resource) => resource.id !== payload
+      );
+    },
+    setResourceDetails(state, payload) {
+      state.resourceDetails = payload;
+    },
+    updateFormData(state, data) {
+      state.resourceDetails = data;
+    },
+    updateResource(state, updatedResource) {
+      const index = state.resources.findIndex(
+        (resource) => resource.id === updatedResource.id
+      );
+      if (index !== -1) {
+        state.resources.splice(index, 1, updatedResource);
```

Comment:
Here we are replacing the resource at `index` with `updatedResource`, right? Can't we just do `state.resources[index] = updatedResource`?

Reply:
yes works